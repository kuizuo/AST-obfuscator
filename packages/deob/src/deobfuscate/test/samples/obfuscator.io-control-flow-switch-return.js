function hi() {
  const _0x4b8693 = _0x44bc
  const _0x526fb7 = {}
  _0x526fb7[_0x4b8693(0x0)] = '5|1|3|0|2|4'
  _0x526fb7[_0x4b8693(0x1)] = _0x4b8693(0x2)
  const _0x20782d = _0x526fb7
  const _0x33c4b2 = _0x20782d[_0x4b8693(0x0)][_0x4b8693(0x3)]('|')
  let _0x313a50 = 0x0
  while ([]) {
    switch (_0x33c4b2[_0x313a50++]) {
      case '0':
        console[_0x4b8693(0x4)](_0x20782d.GEYei)
        continue
      case '1':
        console[_0x4b8693(0x4)](_0x20782d[_0x4b8693(0x1)])
        continue
      case '2':
        console[_0x4b8693(0x4)](_0x20782d[_0x4b8693(0x1)])
        continue
      case '3':
        console[_0x4b8693(0x4)](_0x20782d[_0x4b8693(0x1)])
        continue
      case '4':
        var _0x52ca5a = {}
        _0x52ca5a.x = 0x1
        return _0x52ca5a
      case '5':
        console[_0x4b8693(0x4)](_0x20782d.GEYei)
        continue
    }
    break
  }
}
function _0x44bc(_0x3fc3a6, _0x44bc45) {
  const _0x2004c0 = _0x3fc3()
  _0x44bc = function (_0x4338c0, _0x393a93) {
    _0x4338c0 = _0x4338c0 - 0x0
    const _0x524425 = _0x2004c0[_0x4338c0]
    return _0x524425
  }
  return _0x44bc(_0x3fc3a6, _0x44bc45)
}
function _0x3fc3() {
  const _0x1e397 = [
    'XfgAV',
    'GEYei',
    'Hello\x20World!',
    'split',
    'log',
  ]
  _0x3fc3 = function () {
    return _0x1e397
  }
  return _0x3fc3()
}
hi()
