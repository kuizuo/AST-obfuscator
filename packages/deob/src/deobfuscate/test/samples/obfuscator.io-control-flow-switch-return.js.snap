function hi() {
  console.log("Hello World!");
  console.log("Hello World!");
  console.log("Hello World!");
  console.log("Hello World!");
  console.log("Hello World!");
  return {
    x: 1
  };
}
hi();